pkg/config/sysprep.go

```diff
@@ -67,31 +67,38 @@ func validateAutounattendPresence(dirPath string) error {
 func CreateSysprepDisks(vmi *v1.VirtualMachineInstance) error {
 	for _, volume := range vmi.Spec.Volumes {
 		if volume.Sysprep != nil {
-			if sysprepVolumeHasContents(*volume.Sysprep) {
-				var filesPath []string
+			err := createSysprepDisk(volume)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
 
-				sysprepSourcePath := GetSysprepSourcePath(volume.Name)
+func createSysprepDisk(volume v1.Volume) error {
+	if sysprepVolumeHasContents(*volume.Sysprep) {
```

Comment:
I may read better if this condition remained in the `CreateSysprepDisks`

Reply:
+1.

It would help if you check the volume's "ability" to create a sysprep disk on the loop, and in the generator function, simply create it.

I recommend something like (in the loop's body):
```golang
if volume.Sysprep != nil && sysprepVolumeHasContents(*volume.Sysprep) {
...
}
```

Or better yet: export that if condition to a predicate, and name it accordingly.

Reply:
Also a general good practice with checks like these is to return early.
So instead if indenting everything into this if, just invert the check:
```
if !sysprepVolumeHasContents(*volume.Sysprep) {
  return nil
}
```
But I tend to agree with the previous comments. EIther move it up or put it into a `shouldCreateSysprep(volume)` and put it here. Then you can also drop the nil check from above down into that and simplify the look even further.